metaflow/datastore/datastore_backend.py

```diff
@@ -181,25 +181,26 @@ def list_content(self, paths):
         """
         raise NotImplementedError
 
-    def save_bytes(self, path_and_bytes, overwrite=False):
+    def save_bytes(self, path_and_bytes_iter, overwrite=False):
         """
         Creates objects and stores them in the datastore.
 
         If overwrite is False, any existing object will not be overwritten and
         an error will be returned.
 
-        The objects are specified in the objects dictionary where the key is the
-        path to store the object and the value is a file-like object from which
-        bytes can be read.
+        The objects are specified in an iterator over (key, path) tuples where
+        the key is the path to store the object and the value is a file-like 
+        bject from which bytes can be read.
 
         Parameters
         ----------
-        path_and_bytes : Dict: string -> (RawIOBase or BufferedIOBase, dict)
-            Objects to store; the first element in the tuple is the actual data
-            to store and the dictionary is additional metadata to store. Keys
-            for the metadata must be ascii only string and elements can be
-            anything that can be converted to a string using json.dumps. If you
-            have no metadata, you can simply pass a RawIOBase or BufferedIOBase.
+        path_and_bytes_iter : Iterator[(string, bytes)]
```

Comment:
Type is wrong and comment below does not match. The iterator returns a tuple string, RawIOBase/BufferedIOBase, metdata (which is a dict and optional).